src/en/developer-layers-interfaces.md

```diff
@@ -1,25 +1,60 @@
-Title: Interface layers  
+Title: Interface layers
 
 # Interface layers
 
-Interface layers are perhaps the most misunderstood type of layer, and are
-responsible for the communication that transpires over a relation between two
-applications. This type of layer encapsulates a single “interface protocol”
-and is generally written and maintained by the author of the primary charm
-that provides that interface. However, it does cover both sides (provides and
-requires) of the relation and turns the two-way key-value store that
-are Juju relations under-the-hood into a full-fledged API for interacting with
-charms supporting that interface.
+Interface layers are responsible for the communication that transpires over a
+relation between two applications. This type of layer encapsulates a single
+"interface protocol" and is generally written and maintained by the author of
+the primary charm that provides that interface. However, it does cover both
+sides (provides and requires) of the relation and turns the two-way key-value
+store that are Juju relations under-the-hood into a full-fledged API for
+interacting with charms supporting that interface.
 
 It is important to note that interface layers **do not** actually implement
 either side of the relation. Instead, they are solely responsible for the
 **communication** that goes on over the relation, relying on charms on either
 end to decide what to do with the results of that communication.
 
-Interface layers currently must be written in Python and extend the ReactiveBase
-class, though they can then be __used__ by any language using the built-in CLI
+Interface layers currently must be written in Python and extend the [Endpoint]
+class, though they can then be *used* by any language using the built-in CLI
 API.
 
+[Endpoint]: https://charmsreactive.readthedocs.io/en/latest/charms.reactive.relations.html#charms.reactive.altrelations.Endpoint
+
+## Terminology
+
+Historically, the term "relation" has been used ambiguously when discussing
+Juju, charms, and applications deployed with Juju using charms.  To make things
+more clear and consistent, we use the following terms:
+
+  * "charm" This is the set of code that encapsulates the operational knowledge
+    that manages the life-cycle of an application when deployed.
+
+  * "application" This is a deployed instance of a charm.  It is hosted on a
+    machine, VM, or container, it has a copy of the charm code, and it is
+    managed by a Juju controller.
+
+  * "interface" This is the protocol used by the charm(s) running on two
+    applications to communicate over a relation established between them.
+    This protocol is what interface layers codify.
+
+  * "endpoint" This is the name that a charm uses as a connection point at which
+    a relation can be made between two deployed applications.  An endpoint is
+    defined in a charm's `metadata.yaml` and specify a name, a type (one of
```

Comment:
I'd add a sample `metadata.yaml` with comments on each key/value pair explaining what denotes the interface name, the endpoint name, the endpoint type etc..